src/Entity/Raffle.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace App\Entity;
+
+use App\Exception\NoCommentsToRaffleException;
+use DateTime;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\RaffleRepository")
+ * @ORM\Table(name="raffles")
+ */
+class Raffle implements \JsonSerializable
+{
+    /**
+     * @ORM\Column(type="uuid")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="NONE")
+     *
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="App\Entity\JoindInEvent")
+     * @ORM\JoinTable(name="raffleEvents")
+     *
+     * @var Collection
+     */
+    private $events;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="App\Entity\JoindInUser")
+     * @ORM\JoinTable(name="raffleWinners")
+     *
+     * @var Collection
+     */
+    private $winners;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="App\Entity\JoindInUser")
+     * @ORM\JoinTable(name="raffleNoShows")
+     *
+     * @var Collection
+     */
+    private $noShows;
+
+    /**
+     * @ORM\Column(type="datetime", nullable=false)
+     *
+     * @var DateTime
+     */
+    private $createdAt;
+
+    public function __construct(string $id, Collection $events)
+    {
+        $this->id        = $id;
+        $this->events    = $events;
+        $this->createdAt = new DateTime();
+        $this->winners   = new ArrayCollection();
+        $this->noShows   = new ArrayCollection();
+    }
+
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    public function getEvents(): Collection
+    {
+        return $this->events;
+    }
+
+    public function getCreatedAt(): DateTime
+    {
+        return $this->createdAt;
+    }
+
+    public function getCommentsEligibleForRaffling(): Collection
+    {
+        $comments = new ArrayCollection();
+
+        foreach ($this->events as $event) {
+            foreach ($event->getTalks() as $talk) {
+                foreach ($talk->getComments() as $comment) {
+                    if (false === $this->userAlreadyWonOrIsNoShow($comment)) {
```

Comment:
maybe we could refactor this?
```
$this->userAlreadyWonOrIsNoShow($comment->getUser)
```

and change the method signature for userAlreadyWonOrIsNoShow() to
```
userAlreadyWonOrIsNoShow(JoindInUser $user)
```

Reply:
Great idea, can you open an issue and create a PR?